README.md

```diff
@@ -1,6 +1,66 @@
 Support for detecting and being notified when an object is mutated.
 
+An observable, simply put, is a continuous stream of events over time.
+
+### What's for
+
+You can use this library to process asynchronous streams of data for logging, data binding or custom changes to them. You can think of it as the former TC39 proposal Object.observe() feature that got deprecated.
```

Comment:
I like this, let's clean it up a bit. How about:

Some suggested uses for this library:
- Observe objects for changes, and log when a change occurs
- Optimize for observable collections in your own APIs and libraries instead of diffing
- Implement simple data-binding by listening to streams

You may want to look at the former TC39 proposal `Observe.observe`, which was deprecated.

(We should also link to this if you can)